pdns_exporter/pdns_exporter.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2018 CESNET
+#
+# LICENSE TERMS
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in
+#    the documentation and/or other materials provided with the
+#    distribution.
+# 3. Neither the name of the Company nor the names of its contributors
+#    may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+#
+# ALTERNATIVELY, provided that this notice is retained in full, this
+# product may be distributed under the terms of the GNU General Public
+# License (GPL) version 2 or later, in which case the provisions
+# of the GPL apply INSTEAD OF those given above.
+#
+# This software is provided ``as is'', and any express or implied
+# warranties, including, but not limited to, the implied warranties of
+# merchantability and fitness for a particular purpose are disclaimed.
+# In no event shall the company or contributors be liable for any
+# direct, indirect, incidental, special, exemplary, or consequential
+# damages (including, but not limited to, procurement of substitute
+# goods or services; loss of use, data, or profits; or business
+# interruption) however caused and on any theory of liability, whether
+# in contract, strict liability, or tort (including negligence or
+# otherwise) arising in any way out of the use of this software, even
+# if advised of the possibility of such damage.
+
+import sys
+import pytrap
+import argparse
+import json
+import gzip
+import ipaddress
+from datetime import datetime
+
+"""
+This module reads flow data from TRAP interface and export flow data to JSON and
+store JSON to files on disk. Data in JSON are in format to process in Passive DNS.
+"""
+
+def store_json(json_data):
+    file_name = "pdns_data_" + datetime.now().strftime("%Y%m%d.%H%M%S") + ".json.gz"
+    with gzip.open(file_name, "wb") as output_file:
+        output_file.write(json.dumps(json_data, sort_keys=False).encode('utf-8'))
```

Comment:
To prepare a file in some more secure way, you should store the content into a tmp file and move (`rename()`) to the final destination.

The reason is to have a complete file in the destination dir immediately when it's accessible for readers (e.g. rsync).

This temp dir can be also configurable by parameter.